### ISSUE TITLE
no-unused-prop-types not validating on destructured props 

### ISSUE THREAD

**Issue by jessicaonly** [yannickcr/eslint-plugin-react#2934]
Hi!

We're using `no-unused-prop-types` in our project, but it doesn't seem to be validating when we use it for certain destructured props. for example:

```
const Child = ({usedType, unusedType}) => {
    return (
        <div className="Child">
            <p>{usedType}</p>
        </div>
    );
};

Child.propTypes = {
    usedType: PropTypes.string.isRequired,
    unusedType: PropTypes.string.isRequired, //  I think this should trigger an error, but it doesn't seem to!
};

export default Child;

```

But, if we pass the props in like `const Child = props => { .... } ` , then the error triggers as expected. We thought maybe that the act of it being passed in and destructured in the definition was it being "used", even though it's not actually used inside the component! It would be helpful to have a way to tell if a prop isn't being used inside a component, even if it is being passed into it in this way.

**Comment by ljharb** [comment]
Yes, this is correct - destructuring it is using it.

The eslint rule no-unused-vars is what I’d expect would error on a destructured but unused variable.

**Comment by jessicaonly** [comment]
Got it... thank you! I'll close this out.
